src/kDataFrames/kDataFrameSTL.cpp

```diff
@@ -0,0 +1,609 @@
+kDataFrameSTLIterator<MapType>::kDataFrameSTLIterator(itType it, kDataFrameSTL<MapType> *origin,
+                                                      uint64_t kSize)
+        : _kDataFrameIterator(kSize) {
+    iterator = it;
+    this->origin = origin;
+    this->KD = this->origin->KD;
+}
+
+template<class MapType>
+kDataFrameSTLIterator<MapType>::kDataFrameSTLIterator(const kDataFrameSTLIterator<MapType> &other) :
+        _kDataFrameIterator(other.kSize) {
+    iterator = other.iterator;
+    this->origin = other.origin;
+    this->KD = other.KD;
+}
+
+template<class MapType>
+_kDataFrameIterator *kDataFrameSTLIterator<MapType>::clone() {
+    return new kDataFrameSTLIterator(*this);
+}
+
+template<class MapType>
+kDataFrameSTLIterator<MapType> &kDataFrameSTLIterator<MapType>::operator++(int) {
+    iterator++;
+    return *this;
+}
+
+template<class MapType>
+uint64_t kDataFrameSTLIterator<MapType>::getHashedKmer() {
+    //return origin->getHasher()->hash(iterator->first);
+    return iterator->first;
+
+}
+
+template<class MapType>
+string kDataFrameSTLIterator<MapType>::getKmer() {
+    return origin->getkmerDecoder()->ihash_kmer(iterator->first);
+    // return iterator->first;
+}
+
+template<class MapType>
+uint64_t kDataFrameSTLIterator<MapType>::getCount() {
+    return iterator->second;
+}
+
+template<class MapType>
+uint64_t kDataFrameSTLIterator<MapType>::getOrder() {
+    return iterator->second;
+}
+
+template<class MapType>
+bool kDataFrameSTLIterator<MapType>::setOrder(uint64_t count) {
+    iterator->second = count;
+}
+
+template<class MapType>
+bool kDataFrameSTLIterator<MapType>::operator==(const _kDataFrameIterator &other) {
+    return iterator == ((kDataFrameSTLIterator *) &other)->iterator;
+}
+
+template<class MapType>
+bool kDataFrameSTLIterator<MapType>::operator!=(const _kDataFrameIterator &other) {
+    return iterator != ((kDataFrameSTLIterator *) &other)->iterator;
+}
+
+
+template<class MapType>
+kDataFrameSTLIterator<MapType>::~kDataFrameSTLIterator() {
+}
+
+/*
+ **********************
+ *** kDataFrameSTL ***
+ **********************
+ */
+
+template<class MapType>
+kDataFrameSTL<MapType>::kDataFrameSTL(uint64_t ksize) {
+    this->class_name = "STL"; // Temporary until resolving #17
+    this->kSize = ksize;
+    KD = new Kmers(kSize, TwoBits_hasher);
+    endIterator= nullptr;
+    reserve(1000000);
+//    hasher = new wrapperHasher<flat_hash_map<uint64_t, uint64_t>::hasher>(MAP.hash_function(), ksize);
+//    this->MAP = MapType(1000000);
+//    endIterator = new kDataFrameIterator(
+//            (_kDataFrameIterator *) new kDataFrameSTLIterator(MAP.end(), this, kSize),
+//            (kDataFrame *) this);
+    // this->hasher = (new IntegerHasher(ksize));
+}
+
+
+template<class MapType>
+kDataFrameSTL<MapType>::kDataFrameSTL(uint64_t ksize, uint64_t nKmers) {
+    this->class_name = "STL"; // Temporary until resolving #17
+    this->kSize = ksize;
+    KD = new Kmers(kSize, TwoBits_hasher);
```

Comment:
@shokrof Why is the TwoBits_hasher used here instead of integer hashing? TwoBits_hasher does not really hash the kmers, just returns the two-bit representation.

Reply:
Are you using here std::hash instead of the hashing modes in kmerDecoder?